rfcs/0000-api-cr-crowding.md

```diff
@@ -0,0 +1,337 @@
+effort, and is appropriately quarantined from the rest of the code.
+
+However, this would set an unhappy precedent about "temporary" endpoints just for dotcom that
+don't need to coexist with the rest of the app. This flies in the face of the API's philosophy to
+this point, of being an open source app that could, in principle, be used by any transit agency,
+and that adheres to transit standards for the most part.
+
+And the most likely outcome of this "experiment" is that it lingers indefinitely. We don't want a
+"temporary", ad-hoc endpoint perpetually in the API, especially when its functionality overlaps
+with a GTFS standard that will (eventually) be accepted, and we'll want to adopt formally, for
+Transit app, Google Maps, etc.
+
+## Not exposed directly, but joined to other models
+
+Another approach is to not have an `/occupancies` endpoint but still have the model and data,
+either simply stuck in a `Trip`, or `include=occupancies` by a trip.
+
+The API has a precedent for something like this, where it digests `calendar.txt` but doesn't
+expose a `Service` resource or endpoint.
+
+Adding the given `occupancy_status` and `occupancy_percentage` values right into the trip, would
+be a straightforward approach. However, this is contrary to the data model in the current
+proposal, which explicitly rejects adding these as columns to `trips.txt` or `stop_times.txt`. The
+reasoning given is that a given trip could have different occupancy predictions for different date
+spans or days of the week. While we could handle this in the API via giving trips a _list_ of
+occupancies, this isn't feasible in `trips.txt` in GTFS. And so in the name of forward
+compatibility with the proposal once it lands, we shouldn't do it that way now.
+
+The GTFS-compatible way of attaching a list of occupancies to a trip in the API would be to still
+have the `Occupancy` resource, and allow `trips` to `include=occupancies`. I'd be open to this
+idea, if dotcom thought it would be the most natural way to work with the data, though I'm
+hesitant about it because I'm not exactly sure of the joining logic here, given the data model
+issues mentioned before. Should we validate date range and days of the week as part of the join?
+My gut feeling is that an isolated `/occupancies` endpoint would be simpler for us to build and
+maintain while the proposal is being worked on. Having to work with occupancies via trips also
+seems like it would couple client implementations between their standard trips and this
+non-standard feature more.
+
+# Prior art
+
+[prior-art]: #prior-art
+
+n/a
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+- What filters and includes does dotcom absolutely need on the `/occupancies` endpoint, to do
+  their work? Can they get by with a simple, unfiltered `/occupancies` with the data and the trip
+  ID in it?
+- Would dotcom strongly prefer retrieving `occupancies` via an `include` on a trip instead?
```

Comment:
It sounds nice but I think it's ok if we don't have that.

Reply:
"sounds nice" you say? Well, /gif you got it, dude. 👍